app/amplience-api/index.ts

```diff
@@ -71,28 +165,63 @@ export class AmplienceAPI {
         this.clientReadyResolve()
     }
 
+    /**
+     * Fetch content from Dynamic Content in batch.
+     * @param args A list of IDs or keys to fetch.
+     * @param params Options for fetch.
+     * @returns Content or errors returned from Dynamic Content.
+     */
     async fetchContent(args: IdOrKey[], params: FetchParams = {}) {
         await this.clientReady
 
-        if (params && !params.locale) {
-            params.locale = 'en-US'
-        }
+        params = addDefaultParams(params)
 
         const client = params?.client ?? this.client
-
-        delete params.client
         const chunks = chunk(args, 12)
 
-        let responses = await Promise.all(chunks.map(async (arg) => (await client.getContentItems(arg, params)).responses))
+        let responses = await Promise.all(
+            chunks.map(
+                async (arg: IdOrKey[]) =>
+                    (await client.getContentItems(arg, getClientParams(params))).responses
+            )
+        )
 
-        return flatten(responses).map((response) => {
+        const items = flatten(responses).map((response) => {
             if ('content' in response) {
                 return response.content
             }
             return response.error
         })
+
+        await this.defaultEnrich(items, params)
```

Comment:
Should probably be hardened so this doesn't do anything to content with error response.

Reply:
Done.